### PR TITLE
Removed unrelated permissions and form alterations.

### PR DESCRIPTION
**Pull request by ivangrynenko**
1. Removed external permissions that are not direct dependency of this module.
2. Applied short array syntax.

### CODE REVIEW

File: vision6.module [hunk 1]
```diff
@@ -485,14 +485,6 @@ function vision6_form_node_form_alter(&$form, &$form_state, $form_id) {
   // spec.
   drupal_add_js(drupal_get_path('module', 'vision6') . '/js/vision6.js', 'file');
 
-  if (user_access('access vision6 publish') && !user_access('publish revisions')) {
```
Comment (ivangrynenko) [inline vision6.module:488]: Vision6 module should not rely on permissions of any other module. The 'publish revisions' is not Drupal core permission and not this module permission.

File: vision6.module [hunk 2]
```diff
@@ -608,36 +600,26 @@ function vision6_workbench_moderation_transition($node, $previous_state, $new_st
 function _vision6_node_beforesave($node) {
   global $base_url;
 
-  $wrapper = entity_metadata_wrapper('node', $node);
+  try {
```
Comment (ivangrynenko) [inline vision6.module:603]: `entity_metadata_wrapper` requires using try/catch to avoid WSOF of unhandled exception.

File: vision6.module [hunk 3]
```diff
@@ -608,36 +600,26 @@ function vision6_workbench_moderation_transition($node, $previous_state, $new_st
...
-  if (!user_access('publish revisions')) {
-    if (!is_null($node->nid) && !empty($node->original)) {
-      $node->current_status = $node->original->status;
-      $node->current_title = $node->original->title;
-      $node->current_comment = $node->original->comment;
-      $node->current_promote = $node->original->promote;
-      $node->current_sticky = $node->original->sticky;
-      $node->revision_moderation = 1;
-    }
+  if (module_exists('workbench_moderation') && ($node->workbench_moderation) && $node->workbench_moderation['current']->state != 'published') {
```
Comment (ivangrynenko) [inline vision6.module:615]: Flattened the multiple if statements.

File: vision6.module [hunk 4]
```diff
@@ -608,36 +600,26 @@ function vision6_workbench_moderation_transition($node, $previous_state, $new_st
...
-  else {
-    if (!$wrapper->status->value()) {
-      return;
-    }
+  elseif (!$wrapper->status->value()) {
+    return;
   }
 
-  if (isset($wrapper->field_vis6_published_as_synd)) {
+  if (!empty($node->field_vis6_published_as_synd)) {
```
Comment (ivangrynenko) [inline vision6.module:622]: Use of $wrapper when the property of an object is not available will cause the PHP Fatal error. Using $node object instead.

**Review by ivangrynenko — COMMENTED** [review]
Apart from the code where I commented, the remaining is code style change only.

**Review by sonnykt — APPROVED** [review]

**Pull request merged**